tests/5.1/assume/test_assume_contains.c

```diff
@@ -0,0 +1,49 @@
+//===--- test_assume_contains.c ---------------------------------------------===//
+//
+//  OpenMP API Version 5.1 Aug 2021
+//
+//  The specification states that "when the contains clause appears on an assume
+//  directive, the application proviedes a hint that constructs that match the
+//  listed directive names are likely to be encountered in the scope of the
+//  assume directive." In this test, we use "target parallel for" as the
+//  directive-name for the contains clause.
+//
+////===----------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+#include <math.h>
+
+#define N 1024
+
+int errors, i;
+
+int test_assume_contains() {
+    int arr[N];
+    for(i = 0; i < N; i++){
+        arr[i] = i;
+    }
+    #pragma omp assume contains(parallel)
```

Comment:
Why not contains(target, parallel, for) ?

Reply:
Good call, originally had them as the combined directive but spec states:

> An encountered construct matches the directive name if it is a combined or composite construct for which a constituent construct has the same name as one of the specified directive names

As well as

>  A directive-name that appears in an absent or contains clause may not be a combined or composite directive name.

So I am unsure whether this means a combined construct would work... I'm assuming this means that if a combined construct is explicity listed in the spec (such as #omp target parallel for), then it is allowed.

I ended up just splitting up just using parallel, but **either way, your recommendation would work**